<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditGlobalVariablesDialog</class>
 <widget class="QDialog" name="EditGlobalVariablesDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>837</width>
    <height>446</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Global variables</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/resources.qrc">
    <normaloff>:/topcased_icons/GlobalVariables_24.png</normaloff>:/topcased_icons/GlobalVariables_24.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="1">
    <widget class="Gui::EditDatatypeNameWidget" name="editDatatypeNameWidget" native="true"/>
   </item>
   <item row="1" column="3">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Initial value</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
     <property name="buddy">
      <cstring>initialValueTextEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Type</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Name</string>
     </property>
     <property name="buddy">
      <cstring>varNameLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="5">
    <widget class="Gui::EditListWidget" name="editListWidget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="varNameLineEdit"/>
   </item>
   <item row="0" column="2" rowspan="2">
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="5">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="1" column="4">
    <widget class="QPlainTextEdit" name="initialValueTextEdit">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item row="0" column="3" colspan="3">
    <widget class="QCheckBox" name="persistentCheckBox">
     <property name="text">
      <string>Persistent across restarts (i.e. datastore)</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::EditListWidget</class>
   <extends>QWidget</extends>
   <header>Gui/EditListWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Gui::EditDatatypeNameWidget</class>
   <extends>QWidget</extends>
   <header>Gui/EditDatatypeNameWidget.h</header>
   <container>1</container>
   <slots>
    <signal>datatypeEdited()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>varNameLineEdit</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>varNameLineEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>50</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editDatatypeNameWidget</sender>
   <signal>datatypeEdited()</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>228</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>persistentCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>467</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>834</x>
     <y>119</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>initialValueTextEdit</sender>
   <signal>textChanged()</signal>
   <receiver>EditGlobalVariablesDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>638</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>835</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotAdd()</slot>
  <slot>slotRemove()</slot>
  <slot>slotMoveUp()</slot>
  <slot>slotMoveDown()</slot>
  <slot>slotSelectedRowChanged()</slot>
  <slot>slotEdited()</slot>
 </slots>
</ui>
